src/Queue/SearchProcessor.php

```diff
@@ -55,8 +58,25 @@ public function __construct(EntityManagerInterface $entityManager, EventDispatch
     public function process(Message $message, Context $session)
     {
         $jsonDecoder = new JsonDecoder(true);
+        /* @var ProcessMessage $processMessage*/
         $processMessage = $jsonDecoder->decode($message->getBody(), ProcessMessage::class);
 
+        // Clean up: find all search that links back to a give source and remove them before sending new index event.
+        // This is done even if the search below is an zero-hit.
+        if (VendorState::DELETE_AND_UPDATE === $processMessage->getOperation()) {
+            $sourceRepos = $this->em->getRepository(Source::class);
+            /* @var Source $source */
+            $source = $sourceRepos->findOneBy([
+                'matchId' => $processMessage->getIdentifier(),
+                'matchType' => $processMessage->getIdentifierType(),
+            ]);
+            $searches = $source->getSearches();
```

Comment:
What if source is null?

Reply:
It's never null at this point... if it is we have an corrupted database

Reply:
Fixed